src/test/rustdoc/intra-doc-link-generic-params.rs

```diff
@@ -0,0 +1,18 @@
+// ignore-tidy-linelength
+
+#![crate_name = "foo"]
+
+//! Here's a link to [`Vec<T>`] and one to [`Box<Vec<Option<T>>>`].
+//!
+//! And what about a link to just [`Option`] and, with the generic, [`Option<T>`]?
+//!
+//! Now let's test a trickier case: [`Vec::<T>::new`], or you could write it as [`Vec::<T>::new()`].
+//! And what about something even harder? That would be [`Vec::<Box<T>>::new()`].
+
+// @has foo/index.html '//a[@href="https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html"]' 'Vec<T>'
+// @has foo/index.html '//a[@href="https://doc.rust-lang.org/nightly/alloc/boxed/struct.Box.html"]' 'Box<Vec<Option<T>>>'
+// @has foo/index.html '//a[@href="https://doc.rust-lang.org/nightly/core/option/enum.Option.html"]' 'Option'
+// @has foo/index.html '//a[@href="https://doc.rust-lang.org/nightly/core/option/enum.Option.html"]' 'Option<T>'
+// @has foo/index.html '//a[@href="https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html#method.new"]' 'Vec::<T>::new'
```

Comment:
This test is also not doing anything. If you want to check the link generated is correct, you could change the text to tell them apart: `or you could write it [with parentheses][Vec::<T>::new()].`